report/src/main/webapp/app/components/keyboardShortcuts.directive.js

```diff
@@ -208,13 +226,54 @@ define(['angularAMD', 'userSettingsService'], function (angularAMD) {
         }
       }
 
-      function toggleNextTest(currentTest) {
-        $(currentTest).nextAll('.aside-report:not(.is-hidden)')
-        .first()
-        .find('.test-name')
-        .click();
+      function toggleNextTestItem(currentTest) {
+        $(currentTest).nextAll(
+          '.aside-report:not(.is-hidden)')
+          .first()
+          .find('.test-url')
+          .first()
+          .click();     
       }
 
+      function togglePrevTestItem(currentTest) {
+        $(currentTest).prevAll(
+          '.aside-report:not(.is-hidden)')
+          .first()
+          .find('.test-url')
+          .last()
+          .click();
+      }
+
+      function clickOnTab() {
+        var currentTab = userSettingsService.getLastTab();
+        var nextTestTabs = $('.nav-tabs').children();
+        for(var i=0;i <nextTestTabs.length; i++) {
+          if($(nextTestTabs[i]).text().replace(/\s/g, '') === currentTab) {
+            $(nextTestTabs[i]).find('a').click();
+            currentTab = null;
+          }
+        }
+      }
+
+      var mutationObserver = new MutationObserver(callback);
+
+      function callback(mutList) {
+        var finished = false;
+        mutList.forEach(function(mut) {
```

Comment:
How about using `Array.prototype.find()` insead of `forEach`? You wouldn't have to do this nasty `finished` guard and finish `forEach` loop (better performance).